pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -145,6 +147,15 @@ func validateMinMaxScale(annotations map[string]string) *apis.FieldError {
 			Paths:   []string{MaxScaleAnnotationKey, MinScaleAnnotationKey},
 		})
 	}
+	// Don't verify max scale annotation if default MaxScale is set, since that will be the actual
+	// max scale calculated by ScaleBounds(...).
+	exceedsMaxScaleLimit := config.MaxScaleLimit > 0 && (max == 0 || max > int64(config.MaxScaleLimit))
```

Comment:
Does this mean if DefaultMaxScale is 5 and MaxScaleLimit is 10 I can set max-scale annotation to 1000? That doesn’t seem right..

I think if a default max scale is set then I can set 0 (because that’s then the default which we already validate is below limit), but any value >0 must still be lower than limit

Reply:
@julz hmm I think I still don't get why we would allow maxScale = 0 when DefaultMaxScale and MaxScaleLimit is set, but not allow maxScale > MaxScaleLimit when DefaultMaxScale is set?